docs/DeveloperGuide.adoc

```diff
@@ -154,89 +145,86 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+** Cons: May have performance issues in terms of speed of the programme.
 
-[NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+* **Alternative 2:** Save in memory the result produced by the Analyser objects, and update result upon modification of
+list of Passwords (e.g. new passwords added/ passwords deleted.)
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+** Pros: Performance of programme will be a lot faster.
 
-image::UndoRedoState4.png[]
+** Cons: Hard to implement. Have to keep track of state of the list of Password objects and check if the list has been modified
+from the last time they were analysed.
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
-
-image::UndoRedoState5.png[]
+// end::undoredo[]
 
-The following activity diagram summarizes what happens when a user executes a new command:
+// tag::dataencryption[]
+=== Data Encryption
 
-image::CommitActivityDiagram.png[]
+==== Initialization and Validation
 
-==== Design Considerations
+All data files of SecureIT are encrypted with a master password set by the user. The initialization and validation of the master password is handled by `TestStorage`.
 
-===== Aspect: How undo & redo executes
+The following sequence diagram shows how the master password is initialized when the user uses the app for the first time, and validated for subsequent uses of the app.
 
-* **Alternative 1 (current choice):** Saves the entire address book.
-** Pros: Easy to implement.
-** Cons: May have performance issues in terms of memory usage.
-* **Alternative 2:** Individual command knows how to undo/redo by itself.
-** Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
-** Cons: We must ensure that the implementation of each individual command are correct.
+image::InitPasswordSequenceDiagram.png[]
 
```

Comment:
We think it would be helpful to have an activity diagram to summarise the feature.